### PR TITLE
Create sidebar

### PR DESCRIPTION
**Pull request by mikiMeerson**

### CODE REVIEW

**Comment by tshuva** [comment]
You have the privilege right now to create the project in type script - I highly recommend you'll do it (https://feralamillo.medium.com/create-react-app-typescript-eslint-and-prettier-699277b0b913 ) is a great guide

File: src/App.js [hunk 1]
```diff
@@ -1,15 +1,12 @@
 import React, {Component} from "react";
-import NavBar from './components/NavBar';
+import Main from './components/main';
+
 import './App.css'
 
 class App extends Component {
```
Comment (tshuva) [inline src/App.js]: change to functional component

if the component will be stateless you can remove the return part and jest do 
```export default ()=> ( <Main />)```

File: src/components/NavBar.js [hunk 2]
```diff
@@ -1,31 +1,63 @@
-import React from "react";
-import {BrowserRouter as Router, Link} from "react-router-dom";
+import React, {useState} from "react";
+import {Link} from "react-router-dom";
 import { faCaretDown } from "@fortawesome/free-solid-svg-icons";
```
Comment (tshuva) [inline src/components/NavBar.js]: import { faCaretDown ,faQuestionCircle ,faHammer ,faTools  } from "@fortawesome/free-solid-svg-icons";

File: src/components/NavBar.js [hunk 3]
```diff
@@ -1,31 +1,63 @@
...
+import { faQuestionCircle } from "@fortawesome/free-solid-svg-icons";
+import { faHammer } from "@fortawesome/free-solid-svg-icons";
+import { faTools } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
 import './NavBar.css';
 import OfekLogo from '../assets/images/OfekLogo.png';
 
-const NavBar = () => {
+const NavBar = props => {
```
Comment (tshuva) [inline src/components/NavBar.js]: Destructuring props like that:
```const NavBar = ({MinMaxMain ,mini})=> {...```

than use them neutral in the code

File: src/App.js [hunk 4]
```diff
@@ -1,15 +1,12 @@
 import React, {Component} from "react";
```
Comment (MrShtrahman) [inline src/App.js:1]: In react, from version 17.0.0 and above, `import react from 'React'` is no longer needed for components to render :)

(Of course, named imports like `useState` are needed as in every other module...)

Comment (MrShtrahman) [inline src/App.js]: From react 17.0.0, `import react from 'React'` is no longer needed for components to render

(Of course, named imports like `useState` are still necessary)

File: src/components/main.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import PyBuild from './Tools/pybuild';
+import CMDWord from './Tools/cmd-word';
+import DTEFileCreator from './Tools/dte-file-creator';
+import MemMap from './Tools/memmap';
+import MissShob from './Tools/miss-shob';
+
+const Main = () => {
+
+  const [minimizeSideBar, setMinimizeSideBar] = useState(true);
+  const MinMaxMain = () => {
```
Comment (MrShtrahman) [inline src/components/main.js]: This name is not that clear...
I personally like to add `toggle` to functions that toggle stuff, something like `toggleMainWidth`, but it's up to you

File: src/components/NavBar.js [hunk 6]
```diff
@@ -1,31 +1,63 @@
...
+    
+
+    const toggleSideBar = () => {
+        props.MinMaxMain();
+        if(props.mini) document.getElementById("sidebar").style.width = "250px";
```
Comment (tshuva) [inline src/components/NavBar.js]: a.    document.getElementById("sidebar").style.width  = props.mini ? "250px" : "85px" ;

b   we don't want to use 'document' in react code, if you desire to change the style of a component do it in the `style` default prop ,
 https://reactjs.org/docs/dom-elements.html#style ( you can read the whole file to understand more about dom manipulation 
or more in general for css in react: https://www.sitepoint.com/react-components-styling-options/

File: src/components/NavBar.js [hunk 7]
```diff
@@ -1,31 +1,63 @@
...
+                <FontAwesomeIcon icon={faHammer} className="side-bar-icon"/>
+                <span className="icon-text">PyBuild</span>
+            </Link>
+            <nav>
+                <div className="dropdown-menu">
+                    <div className="dropdown-menu-item">
+                    <div className="dropdown-container">
+                    <FontAwesomeIcon icon={faTools} className="side-bar-icon" />
+                    <span className="icon-text">Tools</span>
+                    <FontAwesomeIcon icon={faCaretDown} style={{marginLeft: "10px"}}/>
```
Comment (MrShtrahman) [inline src/components/NavBar.js]: Please add to a different class or id rather than using `style={}`

File: src/components/NavBar.js [hunk 8]
```diff
@@ -1,31 +1,63 @@
...
-                <li>
-                    <Link to="/pybuild" className="navItem">PyBuild</Link>
-                </li>
-                <li>
-                    <Link to="/tools" className="navItem"><FontAwesomeIcon icon={faCaretDown} style={{marginRight: "10px"}} />Tools</Link>
-                </li>
-            </ul>
+      <div>
+        <nav id="sidebar" className="sidebar" role="navigation"
+             onMouseOver={() => toggleSideBar()} onMouseOut={() => toggleSideBar()}>
```
Comment (tshuva) [inline src/components/NavBar.js]: toggle is a perfect timing to useState with boolean i think

File: src/components/NavBar.js [hunk 9]
```diff
@@ -1,31 +1,63 @@
...
+                <div className="dropdown-menu">
+                    <div className="dropdown-menu-item">
+                    <div className="dropdown-container">
+                    <FontAwesomeIcon icon={faTools} className="side-bar-icon" />
+                    <span className="icon-text">Tools</span>
+                    <FontAwesomeIcon icon={faCaretDown} style={{marginLeft: "10px"}}/> 
+                    </div>
+                        <div className="dropdown-menu">
+                            <div className="dropdown-menu-item">
+                                <Link to="/cmd-word"><span  className="tool-item">CMD Word</span></Link>
```
Comment (MrShtrahman) [inline src/components/NavBar.js]: Read about how to render an **array** of jsx components. 😃 
It's an awsome property of react, REALLY fucking useful and could save you a lot of time in the future

(Hint: notice that each compoenent is a `span` inside of a `Link`, with the same class name)

Comment (tshuva) [inline src/components/NavBar.js]: before the return :
 links = [
{link :'/cmd-word' , word: 'CMD Word'},
{link :'/dte-file-creator' , word: 'DTE File Creator'},
{link :'/miss-shob' , word: Miss Shob'},
{link :'/memmap' , word: 'MemMap'},
]

here
links.map(link =>  <Link to="/link .link"><span className="tool-item">{{link.word}}</span></Link>)

Comment (tshuva) [reply src/components/NavBar.js]: also can be implemented up in the `'/', 'about' and  'pybuild'`  part

File: src/components/NavBar.js [hunk 10]
```diff
@@ -1,31 +1,63 @@
...
+                                <Link to="/dte-file-creator"><span className="tool-item">DTE File Creator</span></Link>
+                                <Link to="/miss-shob"><span className="tool-item">Miss Shob</span></Link>
+                                <Link to="/memmap"><span className="tool-item">MemMap</span></Link>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </nav>
         </nav>
-    </Router>
```
Comment (tshuva) [inline src/components/NavBar.js:27]: y not router?

File: src/components/NavBar.js (hunk 6 above)
Comment (tshuva) [reply src/components/NavBar.js]: @MrShtrahman pls explain better than me (about css in react in general .. css modules etc.)

File: src/components/NavBar.css [hunk 11]
```diff
@@ -1,76 +1,130 @@
...
+ nav a {
+    color:#FFF;
+    min-width:190px;
+    transition: color 0.3s, transform 0.3s;
+    padding-bottom: 20px;
+    position:relative;
+ }
+ 
 
-  .ofeklogo {
```
Comment (tshuva) [inline src/components/NavBar.css:71]: great - now no more ofek  in the a respectable project

File: src/components/NavBar.js (hunk 3 above)
Comment (mikiMeerson) [reply src/components/NavBar.js]: one of the props is a function so I declared props types above

File: src/App.js (hunk 4 above)
Comment (mikiMeerson) [reply src/App.js]: Throws an error

Comment (MrShtrahman) [reply src/App.js]: Yep, my bad.
Add these to your `eslint.json` file:
```
{
  // ...
  "rules": {
    // ...
    "react/jsx-uses-react": "off",
    "react/react-in-jsx-scope": "off"
  }
}
```

File: src/components/NavBar.js (hunk 3 above)
Comment (MrShtrahman) [reply src/components/NavBar.js]: Even if one or more of your props is a function, you can still destrcut them (and I highly suggest you to do so):

```
interface PropsType {
  toggleMainWidth: () => void;
  mini: Boolean;
}
const NavBar = ({toggleMainWidth, mini}: PropsType) => {
```

Also, please give your interface a more specific name, something such as `NavBarPropsType`

File: src/components/NavBar.js (hunk 8 above)
Comment (mikiMeerson) [reply src/components/NavBar.js]: Happens in parent component

File: formatting-project/src/reportWebVitals.ts [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import { ReportHandler } from 'web-vitals';
```
Comment (tshuva) [inline formatting-project/src/reportWebVitals.ts:1]: you can delete it and remove the reference in  index.ts and package.json (this package not exist  in nexus )

File: src/components/NavBar.tsx [hunk 13]
```diff
@@ -7,22 +6,36 @@ import {
...
 
-interface PropsType {
-  toggleMainWidth(): void;
+interface NavBarPropsType {
+  toggleMainWidth:() => void;
   mini: Boolean;
 }
-const NavBar = (props: PropsType) => {
+const NavBar = ({ toggleMainWidth, mini }: NavBarPropsType) => {
   const SideBarElement: HTMLElement | null = document.getElementById('sidebar');
```
Comment (MrShtrahman) [inline src/components/NavBar.tsx:19]: Please use `useRef` instead of referencing directly to the DOM.

Here's a really good tutorial for this important hook: [useRef tutorial](https://www.youtube.com/watch?v=t2ypzz6gJm0)

File: src/components/NavBar.tsx [hunk 14]
```diff
@@ -7,22 +6,36 @@ import {
...
+    }
+  ];
+
   const toggleSideBar = (mouse: Boolean) => {
     if(mouse) {
-      props.toggleMainWidth();
+      toggleMainWidth();
       if (SideBarElement) {
-        if (props.mini) SideBarElement.style.width = '250px';
+        if (mini) SideBarElement.style.width = '250px';
```
Comment (MrShtrahman) [inline src/components/NavBar.tsx]: `SideBarElement.style.width = mini ? '250px' : '80px';`

File: src/components/NavBar.tsx [hunk 15]
```diff
@@ -7,22 +6,36 @@ import {
...
-  toggleMainWidth(): void;
+interface NavBarPropsType {
+  toggleMainWidth:() => void;
   mini: Boolean;
 }
-const NavBar = (props: PropsType) => {
+const NavBar = ({ toggleMainWidth, mini }: NavBarPropsType) => {
   const SideBarElement: HTMLElement | null = document.getElementById('sidebar');
 
+  const pages =  [
```
Comment (MrShtrahman) [inline src/components/NavBar.tsx]: Great! 
Although I think that putting your pages on a different file will be a better practice (same you did with "tools")

File: src/components/NavBar.js (hunk 9 above)
Comment (mikiMeerson) [reply src/components/NavBar.js]: Tried something else

File: src/components/NavBar.js (hunk 6 above)
Comment (mikiMeerson) [reply src/components/NavBar.js]: changed classes dynamically instead and declared style in css

File: src/components/NavBar.tsx (hunk 14 above)
Comment (mikiMeerson) [reply src/components/NavBar.tsx]: changed it

File: src/components/main.tsx [hunk 16]
```diff
@@ -23,13 +17,9 @@ const Main = () => {
...
       <div className="content" id={mainId}>
         <Switch>
-          <Route exact path="/" component={Home} />
-          <Route path="/about" component={About} />
-          <Route path="/pybuild" component={PyBuild} />
-          <Route path="/cmd-word" component={CMDWord} />
-          <Route path="/memmap" component={MemMap} />
-          <Route path="/dte-file-creator" component={DTEFileCreator} />
-          <Route path="/miss-shob" component={MissShob} />
+          {routes.map((route) => (
```
Comment (MrShtrahman) [inline src/components/main.tsx:37]: Careful! You're losing the `exact` prop of the home route

File: src/components/NavBar/NavBar.tsx [hunk 17]
```diff
@@ -13,13 +13,13 @@ interface NavBarPropsType {
...
 const NavBar = ({ toggleMainWidth, mini }: NavBarPropsType) => {
-  const [sideBarId, setSideBarId] = useState("sidebar-min");
+  const [sideBarId, setSideBarId] = useState('sidebar-min');
 
   const toggleSideBar = (mouse: Boolean) => {
     if (mouse) {
       toggleMainWidth();
-      if (mini) setSideBarId("sidebar-max");
-      else setSideBarId("sidebar-min");
+      if (mini) setSideBarId('sidebar-max');
```
Comment (MrShtrahman) [inline src/components/NavBar/NavBar.tsx:21]: `mini ? setSideBarId('sidebar-max') : setSideBarId('sidebar-min')`

File: src/components/NavBar/NavBar.tsx [hunk 18]
```diff
@@ -13,13 +13,13 @@ interface NavBarPropsType {
   mini: Boolean;
 }
 const NavBar = ({ toggleMainWidth, mini }: NavBarPropsType) => {
-  const [sideBarId, setSideBarId] = useState("sidebar-min");
+  const [sideBarId, setSideBarId] = useState('sidebar-min');
 
   const toggleSideBar = (mouse: Boolean) => {
```
Comment (MrShtrahman) [inline src/components/NavBar/NavBar.tsx]: Please change `mouse` to a more meaningful name (for a boolean variable)

File: src/components/NavBar/NavBar.tsx [hunk 19]
```diff
@@ -13,13 +13,13 @@ interface NavBarPropsType {
   mini: Boolean;
 }
 const NavBar = ({ toggleMainWidth, mini }: NavBarPropsType) => {
-  const [sideBarId, setSideBarId] = useState("sidebar-min");
```
Comment (MrShtrahman) [inline src/components/NavBar/NavBar.tsx]: Why changing the id of your component (which is highly risky and could lead to css issues), when you can add a second css class and change it dynamically?

In you css file, change the ids that control the width to classes, and then combine `sidebar` and `sideBarId` toghther (Read about [Template Literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals))

File: src/components/NavBar/NavBar.tsx (hunk 17 above)
Comment (MrShtrahman) [inline src/components/NavBar/NavBar.tsx]: `mini ? setSideBarId('sidebar-max') : setSideBarId('sidebar-min')`

File: src/assets/routes.ts [hunk 20]
```diff
@@ -0,0 +1,40 @@
+import Home from '../components/Home';
+import About from '../components/About';
+import PyBuild from '../components/Tools/pybuild';
+import CMDWord from '../components/Tools/cmd-word';
+import DTEFileCreator from '../components/Tools/dte-file-creator';
+import MemMap from '../components/Tools/memmap';
+import MissShob from '../components/Tools/miss-shob';
+
+const routes = [
```
Comment (MrShtrahman) [inline src/assets/routes.ts]: (Not critical, but VERY good practice):

Notice that `react-router` has `RouteProps` as a named export, which you can import.

You can define `routes` to be of type `RouteProps[]`, and then have better control over which elements you pass to each `<Route />` component as props.

(Of course, if you do that, you'd have to change `link` to `path` and `comp` to `component`, but it would make more sense to anyone new reading this code)

File: package.json [hunk 21]
```diff
@@ -12,18 +13,30 @@
     "@testing-library/jest-dom": "^5.11.9",
     "@testing-library/react": "^11.2.5",
     "@testing-library/user-event": "^12.8.1",
+    "@types/jest": "^26.0.20",
```
Comment (MrShtrahman) [inline package.json]: Please run the following command:
`npm i -D @types/jest @types/node @types/react @types/react-dom`.
This will automatically move all of these modules to the `devDependencies` section.

In general, a good approach is to ask yourself the following question: "Am I going to **import** something explicitly from this module? If yes, it's a dependency. Otherwise -  a devDependency. 

In this case, you are never going to import anything directly from any `@types` module, so it should be a devDep

File: tsconfig.json [hunk 22]
```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "target": "es5",
```
Comment (MrShtrahman) [inline tsconfig.json:3]: You can change also target to "esnext"

File: src/components/NavBar/NavBar.tsx (hunk 19 above)
Comment (mikiMeerson) [reply src/components/NavBar/NavBar.tsx]: Added a second class

File: formatting-project/src/App.tsx [hunk 23]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
```
Comment (tshuva) [inline formatting-project/src/App.tsx:1]: i guess that this file can be deleted

File: src/components/NavBar/NavBar.tsx [hunk 24]
```diff
@@ -0,0 +1,62 @@
...
+      toggleMainWidth();
+      setSideBarWidthClass(mini ? 'sidebar-max' : 'sidebar-min');
+
+    }
+  };
+
+  return (
+    <div>
+      <nav
+        className={ ["sidebar", sideBarWidthClass].join(' ') }
```
Comment (tshuva) [inline src/components/NavBar/NavBar.tsx]: i think you can also do \`sidebar ${sideBarWidthClass}\`

(generally -when you want to string a  variable you can with the backtick and ${} )

File: src/components/NavBar/NavBar.tsx [hunk 25]
```diff
@@ -0,0 +1,62 @@
...
+
+interface NavBarPropsType {
+  toggleMainWidth: () => void;
+  mini: Boolean;
+}
+const NavBar = ({ toggleMainWidth, mini }: NavBarPropsType) => {
+  const [sideBarWidthClass, setSideBarWidthClass] = useState('sidebar-min');
+
+  const toggleSideBar = (isMouseEvent: Boolean) => {
+    if (isMouseEvent) {
```
Comment (tshuva) [inline src/components/NavBar/NavBar.tsx:26]: i think it a good idea to implement some kind of global store here.
## tl:dr - can be bad
             the reason behind this is that you don't really want two different sittings for when you open the sidebar
             (mainWidthClass and sideBarWidthClass) cause what will happen if you need to change more components
             with it, or bug will happen or whatever(I have  another like 3 scenarios of bad things that can happen) 
##
it will be better if you implement redux(if the project will be big and will need global store) or use context API(also a global store but smaller ) and store whether the sidber is ***open*** (not max) or not

than in every relevant component's files(tsx and css) pull from the store the current state(open/close) and add it to the class name


so like for example in here you'll have in the css file 
```
sidebar, open {.....}
sidebar, close {.....}
```
and in the tsx :
```
const stateOfSideBar = gettingDataFromStore(stateOfSidebar)
<nav classname \`sidebar ${stateOfSideBar }\`
```

Comment (tshuva) [reply src/components/NavBar/NavBar.tsx:26]: i have new computer so new keyboard so i like to type- sorry

Comment (tshuva) [reply src/components/NavBar/NavBar.tsx:26]: also - link to redux :
https://blog.logrocket.com/why-use-redux-reasons-with-clear-examples-d21bffd5835/
(if you like to use: https://redux.js.org/introduction/getting-started)
and context: 
https://reactjs.org/docs/context.html

File: tsconfig.json (hunk 22 above)
Comment (tshuva) [reply tsconfig.json:3]: also there a configuration somewhere that Mr S will tell us that can sexify the imports
so instead of using "../../../../main" you willdo "components/main" 

also you doctor @MrShtrahman can add our eslint configurations etc.

File: src/components/NavBar/NavBar.tsx (hunk 19 above)
Comment (MrShtrahman) [reply src/components/NavBar/NavBar.tsx]: Kapara on your life

File: src/components/main.tsx [hunk 26]
```diff
@@ -1,24 +1,26 @@
...
   return (
     <Router>
       <NavBar toggleMainWidth={toggleMainWidth} mini={minimizeSideBar} />
-      <div className="content" id={mainId}>
+      <div className={ ["content",mainWidthClass].join(' ') } >
         <Switch>
           {routes.map((route) => (
-            <Route key={route.link} path={route.link} component={route.comp}/>
+            <Route key={route.link}
+                   exact path={route.link}
```
Comment (MrShtrahman) [inline src/components/main.tsx]: Not quite what I meant. In this way, every one of your routes will be exact, which is not that optimal.
There are a couple of possible solutions:
1. You could take the `home` route outside of the array as an "exception", and write this one "hard-coded". The problem is... you know... having something hard-coded.
2. You could add another parameter, "exact" to `routes`, and then write `exact={route.exact}`. The problem here is that you'll have to write `exact: false` in each one...

I tend to go more with the first solution, but it's up to you. Anyway, please change it.

File: package.json [hunk 27]
```diff
@@ -13,14 +13,11 @@
...
     "@testing-library/user-event": "^12.8.1",
-    "@types/jest": "^26.0.20",
-    "@types/node": "^14.14.31",
-    "@types/react": "^17.0.3",
-    "@types/react-dom": "^17.0.1",
     "@types/react-router-dom": "^5.1.7",
     "font-awesome": "^4.7.0",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
+    "react-route-props": "^1.0.4",
```
Comment (MrShtrahman) [inline package.json]: No need for that one at all 😆 
`react-router` has already all the props you need!

Delete this module and in `routes.ts` add the following:
`import { RouteProps } from 'react-router'`

File: src/components/NavBar/NavBar.tsx [hunk 28]
```diff
@@ -13,21 +13,20 @@ interface NavBarPropsType {
   mini: Boolean;
 }
 const NavBar = ({ toggleMainWidth, mini }: NavBarPropsType) => {
-  const [sideBarId, setSideBarId] = useState('sidebar-min');
+  const [sideBarWidthClass, setSideBarWidthClass] = useState('sidebar-min');
```
Comment (MrShtrahman) [inline src/components/NavBar/NavBar.tsx:23]: @tshuva hates positive comments, but I don't really care: great job!

File: src/components/NavBar/NavBar.tsx [hunk 29]
```diff
@@ -13,21 +13,20 @@ interface NavBarPropsType {
...
+
     }
   };
 
   return (
     <div>
       <nav
-        id={sideBarId}
-        className="sidebar"
+        className={ ["sidebar", sideBarWidthClass].join(' ') }
```
Comment (MrShtrahman) [inline src/components/NavBar/NavBar.tsx]: Please change to ``className={`sidebar ${sideBarWidthClass}`}``, no need for arrays here

File: src/components/main.tsx [hunk 30]
```diff
@@ -1,24 +1,26 @@
...
     setMinimizeSideBar(!minimizeSideBar);
-    setMainId(minimizeSideBar ? "main-min" : "main-max");
+    setMainWidthClass(minimizeSideBar ? 'main-min' : 'main-max');
   };
 
   return (
     <Router>
       <NavBar toggleMainWidth={toggleMainWidth} mini={minimizeSideBar} />
-      <div className="content" id={mainId}>
+      <div className={ ["content",mainWidthClass].join(' ') } >
```
Comment (MrShtrahman) [inline src/components/main.tsx]: Please change to ``className={`content ${mainWidthClass}`}``

File: src/components/NavBar/NavBar.tsx [hunk 31]
```diff
@@ -0,0 +1,70 @@
+import { useState } from 'react';
```
Comment (tshuva) [inline src/components/NavBar/NavBar.tsx:1]: you can change the file name to index.tsx and the imports will be : ` import { NavBar} from  'src/components/NavBar'`

File: src/components/NavBar/NavBar.tsx [hunk 32]
```diff
@@ -0,0 +1,70 @@
...
+      setSideBarWidthClass(mini ? 'sidebar-max' : 'sidebar-min');
+    }
+  };
+
+  return (
+    <div>
+      <nav
+        className={`sidebar ${sideBarWidthClass}`}
+        role="navigation"
+        onMouseOver={() => toggleSideBarWidth(true)}
```
Comment (tshuva) [inline src/components/NavBar/NavBar.tsx:37]: you do realize that when you are passing false nothing happens and you can just not write  onFocus and onBlur ( i understand that probably you had good reason but if not..)

File: src/components/NavBar/NavBarToolItem.tsx [hunk 33]
```diff
@@ -0,0 +1,14 @@
+import { Link } from 'react-router-dom';
```
Comment (MrShtrahman) [inline src/components/NavBar/NavBarToolItem.tsx]: I don't think you need to define a component for this one 🤔 
If its purpose is only to be a `<span />` within a `<Link />`, you could just write it in `NavBarTools` instead

File: src/components/NavBar/CircleMenu.css [hunk 34]
```diff
@@ -0,0 +1,276 @@
+body {
```
Comment (MrShtrahman) [inline src/components/NavBar/CircleMenu.css]: Wish I knew css good enough in order to review it...

File: src/components/NavBar/PlatformsMenu.tsx [hunk 35]
```diff
@@ -0,0 +1,48 @@
...
+      onKeyDown={() => switchPlatform('F15')}
+    >
+      <FontAwesomeIcon icon={faSpaceShuttle} />
+    </div>
+    <div
+      className="menu-item f16-style"
+      role="button"
+      tabIndex={0}
+      title="f16"
+      onClick={() => switchPlatform('F16')}
```
Comment (MrShtrahman) [inline src/components/NavBar/PlatformsMenu.tsx]: Maybe add platforms to an enum

File: src/components/main.tsx [hunk 36]
```diff
@@ -3,19 +3,36 @@ import { HashRouter as Router, Route, Switch } from 'react-router-dom';
...
+  const [platform, setPlatform] = useState(platforms[0]);
+
   const toggleMainWidth = () => {
     setMinimizeSideBar(!minimizeSideBar);
     setMainWidthClass(minimizeSideBar ? 'main-min' : 'main-max');
   };
 
+  const switchPlatform = (platformID: string) => {
+    const newPlatform = platforms.find((p) => p.id === platformID);
+    if (newPlatform) {
```
Comment (MrShtrahman) [inline src/components/main.tsx]: `newPlatform && setPlatform(newPlatform);`

File: src/components/NavBar/NavBarTools.tsx [hunk 37]
```diff
@@ -1,16 +1,22 @@
...
+const NavBarTools = ({ platform }: ToolsProps) => (
   <div className="dropdown-menu">
     <div className="dropdown-menu-item">
-      {tools.map((tool) => (
-        <Link to={tool.link} key={tool.display}>
-          <span className="tool-item">{tool.display}</span>
-        </Link>
-      ))}
+      {tools
+        .filter((tool) => tool.tags.includes(platform))
```
Comment (MrShtrahman) [inline src/components/NavBar/NavBarTools.tsx]: You're passing `platform` all the way from `<main />` to here and to other "child" components - it's time to introduce redux! 😄 




(But in the next PR though..... please)

File: src/components/NavBar/CircleMenu.css (hunk 34 above)
Comment (mikiMeerson) [reply src/components/NavBar/CircleMenu.css]: Dont even try

File: src/components/NavBar/PlatformsMenu.tsx (hunk 35 above)
Comment (mikiMeerson) [reply src/components/NavBar/PlatformsMenu.tsx]: Did something else

**Comment by MrShtrahman** [comment]
Ok, so this PR is happily getting really close to being merged (Unless @tshuva unleashes like a GAZILLIION things that I missed, in which case... well... good luck).

One last tiny thing that I missed: in `<NavBar />`  you're defning `mini: Boolean`, please replace to `boolean`, these are two very different types! (Long story short, `Boolean` is a function, `boolean` is the primitive type you want)

There are a few things needed to be added to your code after the merge (preferably before the hackaton), I'll list them here for you to consider:

1. You need to decide about your naming standarts for components, you can't have both `NavBar.tsx`, `main.tsx` and `cmd-word.tsx`. I personally prefer `NavBar, Main, CmdWord` etc... but it's your project- you decide. Anyway, please let all of them be consistent.
2. I think you can delete `formatting-project` already 😆 
3.  You have a lot of css. That's not a bad thing on its own, but when your project will grow larger (and literally two whole departments will work on it simultaneously), mix-ups and css conflicts would be inevitable. The solution is using [css modules](https://spin.atomicobject.com/2020/06/22/css-module-typescript/). We do it in our Gen-B, and **this shit saves lives**. Add it as long as your project is still small.
4. As mentioned before, [redux!!!](https://react-redux.js.org/). Should I even convince you?
5. Tests!!! Currently you don't have any logic (only UI), but you will have in the future. [react-testing-library](https://testing-library.com/docs/react-testing-library/intro/) will come to the rescue here, testing both UI and logic.

Regardless, this project looks really good, and I definitely see it succees! Your progress is wonderful, keep up the good work! 🚀

**Comment by tshuva** [comment]
OK sooo apparently i dont know how to use technology (i know shockinggg) so alllot of comment from week ago wasn't written until now, it is possible that ALOT of them are redundant - ill try to delete those 



edit : i found that ctrl+enter doesn't do comment but review and I dont know)

File: src/.eslintrc [hunk 38]
```diff
@@ -0,0 +1,94 @@
+// .eslintrc
```
Comment (tshuva) [inline src/.eslintrc:1]: I am pretty sure that .eslint(and prettierrc ) files need to be in root folder(@MrShtrahman )

also, both of you start with an M and it bummer , i will not tell you to change but im Seriously upset

File: src/components/NavBar/NavBar.tsx [hunk 39]
```diff
@@ -0,0 +1,73 @@
...
+import { Link } from 'react-router-dom';
+import { faCaretDown, faTools } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { pages } from '../../assets/pages';
+
+import './NavBar.css';
+import NavBarTools from './NavBarTools';
+import NavBarLogo from './NavBarLogo';
+
+interface PlatformType {
```
Comment (tshuva) [inline src/components/NavBar/NavBar.tsx:11]: why  PlatformType defined here and not in the enum file?

File: src/components/NavBar/NavBar.tsx (hunk 25 above)
Comment (MrShtrahman) [reply src/components/NavBar/NavBar.tsx:26]: @tshuva So I guess you didn't read my big comment above 😞

File: src/.eslintrc (hunk 38 above)
Comment (MrShtrahman) [reply src/.eslintrc:1]: You actually think i'm going to change "MrShtrahman" becuase you're too lazy **TO TYPE ANOTHER KEYBOARD???**

Comment (MrShtrahman) [reply src/.eslintrc:1]: But yeah, good point about the files being transferred to the root directory.

File: src/components/main.tsx (hunk 16 above)
Comment (tshuva) [reply src/components/main.tsx:37]: delete/ resolve this if its done

File: src/components/NavBar/NavBar.tsx [hunk 40]
```diff
@@ -0,0 +1,73 @@
...
+        onFocus={() => toggleSideBarWidth(false)}
+        onMouseOut={() => toggleSideBarWidth(true)}
+        onBlur={() => toggleSideBarWidth(false)}
+      >
+        <NavBarLogo header={platform.header} />
+        <div className="navbar-items">
+          {pages
+            .filter(
+              (page) =>
+                platform.id === 'home' || page.tags.includes(platform.id)
```
Comment (tshuva) [inline src/components/NavBar/NavBar.tsx:47]: also can be in redux , some sort of relevantPlatforms  variable  instated of this unnecessary filtering

File: package.json [hunk 41]
```diff
@@ -12,18 +13,26 @@
...
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "predeploy": "npm run build",
+    "deploy": "gh-pages -d build",
+    "lint": "eslint --ext .js,.jsx,.ts,.tsx src --color",
+    "format": "prettier --write 'src/**/*.{ts,tsx,scss,css,json}'",
+    "isready": "npm run format && npm run lint && npm run build"
```
Comment (tshuva) [inline package.json:35]: you dont have tsc script (and i thing you should- @dor132 stohk t'ph @MrShtrahman Explain more pls)

File: src/components/NavBar/PlatformsMenu.tsx [hunk 42]
```diff
@@ -0,0 +1,34 @@
...
+
+const PlatformsMenu = ({ switchPlatform }: PlatformsProps) => (
+  <nav className="menu">
+    <input type="checkbox" className="menu-open" id="menu-open" />
+    <label className="menu-open-button" htmlFor="menu-open">
+      <span className="lines line-1" />
+      <span className="lines line-2" />
+      <span className="lines line-3" />
+    </label>
+    {platforms.map((p) => (
```
Comment (tshuva) [inline src/components/NavBar/PlatformsMenu.tsx:17]:
```suggestion
    {platforms.map((platform) => (
```

File: src/components/NavBar/NavBar.tsx [hunk 43]
```diff
@@ -0,0 +1,73 @@
...
+        className={`sidebar ${platform.style} ${sideBarWidthClass}`}
+        role="navigation"
+        onMouseOver={() => toggleSideBarWidth(true)}
+        onFocus={() => toggleSideBarWidth(false)}
+        onMouseOut={() => toggleSideBarWidth(true)}
+        onBlur={() => toggleSideBarWidth(false)}
+      >
+        <NavBarLogo header={platform.header} />
+        <div className="navbar-items">
+          {pages
```
Comment (tshuva) [inline src/components/NavBar/NavBar.tsx:44]: not sure if to this PR but

why not impmant routings that you'll have src(home) than platform than specific page?
like 'https://HUB/F15/Py-Build'

and than the separation will be in the routing level and not here?

Comment (mikiMeerson) [reply src/components/NavBar/NavBar.tsx:44]: The tools can belong to several platforms and I dont want to seperate them this way.
The platforms are only meant for filtering the navbar options (display only).
In other words, each tool should be logically accessible regardless of the user's chosen platform

File: formatting-project/src/App.tsx (hunk 23 above)
Comment (mikiMeerson) [reply formatting-project/src/App.tsx:1]: It holds main

File: src/components/NavBar/NavBar.tsx (hunk 24 above)
Comment (mikiMeerson) [reply src/components/NavBar/NavBar.tsx]: outdated

File: src/components/NavBar.js (hunk 10 above)
Comment (mikiMeerson) [reply src/components/NavBar.js:27]: It's in main

File: src/components/NavBar/NavBar.tsx (hunk 25 above)
Comment (mikiMeerson) [reply src/components/NavBar/NavBar.tsx:26]: Next PR, thank you both

File: src/components/NavBar/NavBar.tsx (hunk 32 above)
Comment (mikiMeerson) [reply src/components/NavBar/NavBar.tsx:37]: eslint demands it for accessibility :(

File: src/components/NavBar/NavBar.tsx (hunk 39 above)
Comment (mikiMeerson) [reply src/components/NavBar/NavBar.tsx:11]: One-time use and no enum file (currently)
Did something else

**Review by MrShtrahman — APPROVED** [review]

File: src/components/NavBar/NavBar.tsx (hunk 25 above)
Comment (tshuva) [reply src/components/NavBar/NavBar.tsx:26]: I write this comment way before the big comment but' next PR
